packages/frontend/src/components/QrReader/index.tsx

```diff
@@ -129,25 +158,40 @@ export default function QrReader({ onError, onScan }: Props) {
     getAllCameras()
   }, [])
 
-  // General handler for scanning results coming from the "jsqr" library.
-  //
-  // Additionally we have checks in place to make sure we're not firing any
-  // callbacks when this React component has already been unmounted.
-  const handleScanResult = useCallback(
-    result => {
-      let unmounted = false
-
-      if (unmounted) {
-        return
+  const handlePasteFromClipboard = useCallback(async () => {
+    try {
+      const result = await processClipBoard(runtime)
+      if (typeof result === 'string') {
+        onScan(result)
+      } else {
+        try {
+          onScan(result.data)
+        } catch (error) {
+          console.error('Error while processing clipboard content:', error)
+        }
       }
-
-      onScan(result.data)
-
-      return () => {
-        unmounted = true
+    } catch (error) {
+      userFeedback({
+        type: 'error',
+        text: `${tx('qrscan_failed')}: ${error}`,
+      })
+    }
+  }, [onScan, tx, userFeedback])
+
+  /**
+   * used to expose the handlePasteFromClipboard
+   * function to the parent component
+   */
+  useImperativeHandle(
+    ref,
+    () => {
+      return {
+        pasteFromClipboard() {
+          handlePasteFromClipboard()
+        },
       }
     },
-    [onScan]
+    [handlePasteFromClipboard]
   )
```

Comment:
Nitpick: given how little `handlePasteFromClipboard` does (especially if you take my other commets into account), I personally wouldn't make it an imperative handle. Instead I'd just completely separate it from the `QrReader` component.